nextcord/interactions/application.py

```diff
@@ -0,0 +1,808 @@
+from .base import (
+    Interaction,
+    InteractionMessage,
+    InteractionResponse,
+    PartialInteractionMessage,
+    _InteractionMessageState,
+)
+
+__all__ = ("ApplicationCommandInteraction", "ApplicationAutocompleteInteraction")
+
+if TYPE_CHECKING:
+    from ..application_command import (
+        BaseApplicationCommand,
+        SlashApplicationSubcommand,
+        SlashOptionData,
+    )
+    from ..message import AllowedMentions
+    from ..state import ConnectionState
+    from ..types.interactions import (
+        ApplicationAutocompleteInteraction as ApplicationAutocompletePayload,
+        ApplicationCommandInteraction as ApplicationCommandPayload,
+        ApplicationCommandInteractionData as InteractionData,
+        ApplicationCommandInteractionDataOption as OptionsPayload,
+    )
+    from ..ui.modal import Modal
+    from ..ui.view import View
+
+MISSING: Any = utils.MISSING
+
+
+class ApplicationCommandInteraction(Interaction):
+    """Represents the interaction for all application commands.
+
+    This interaction is a subclass of :class:`Interaction`.
+
+    .. container:: operations
+
+        .. describe:: x == y
+
+            Checks if two interactions are equal.
+
+        .. describe:: x != y
+
+            Checks if two interactions are not equal.
+
+        .. describe:: hash(x)
+
+            Returns the interaction's hash.
+
+    Attributes
+    ----------
+    application_command: Union[SlashApplicationSubcommand, BaseApplicationCommand]
+        The application command that triggered the interaction.
+    app_command_name: :class:`str`
+        The name of the application command that triggered the interaction.
+    app_command_id: :class:`int`
+        The application command ID that triggered the interaction.
+    options: List[SlashOptionData]
+        The application command options that have been given a value.
+    """
+
+    __slots__: Tuple[str, ...] = (
+        "application_command",
+        "app_command_name",
+        "app_command_id",
+        "options",
+    )
+
+    def __init__(self, *, data: ApplicationCommandPayload, state: ConnectionState) -> None:
+        super().__init__(data=data, state=state)
+
+        self.application_command: Optional[
+            Union[SlashApplicationSubcommand, BaseApplicationCommand]
+        ] = None
+
+    def _from_data(self, data: ApplicationCommandPayload) -> None:
+        super()._from_data(data=data)
+
+        self.data: InteractionData = data.get("data")
+        self.app_command_name: str = self.data["name"]
+        self.app_command_id: Snowflake = self.data["id"]
+
+        options = self.data.get("options")
+        self.options: List[SlashOptionData] = (
+            self._get_application_options(options) if options else []
+        )
+
+    def _set_application_command(
+        self, app_cmd: Union[SlashApplicationSubcommand, BaseApplicationCommand]
+    ) -> None:
+        self.application_command = app_cmd
```

Comment:
This is part of old code not written by me. From searching through the library, by the looks of things, it gets used in `CallbackMixing.invoke_callback_with_hooks()`. Im guessing it serves a purpose, but since its not my code, I've not looked into it